Refactor content method

jonathanyeong opened the pull request:
Break down the content method into smaller pieces. Also tried to move any repeating logic into it's own method. @sarony.

jonathanyeong committed:
Break down the content method in html_factory

This splits the large content method into smaller methods.

jonathanyeong committed:
Write specs for content method

These specs ensure that the sponsors and the callout section exists.

jonathanyeong committed:
Remove order dependency from html_output method

jonathanyeong committed:
Refactor out the section title

This code is replicated over all the sections. Place the logic into one
method which will return a string containing the section title and any
styling related to that section title.

jonathanyeong committed:
Refactor html of posts that have pictures

This html is now placed in one method.

sarony commented on lib/html_factory.rb:
```diff
@@ -675,29 +675,21 @@ def body_end
+    string << content_list_content(newsletter_content[:content_lists])
+    string << sponsor_content(newsletter_content[:sponsors])
+    string << callout_content(newsletter_content[:callouts])
+  end
```
This looks SO MUCH BETTER. How do you feel about it?

jonathanyeong replied:
It feels so good! That beautiful 4 line method.

sarony commented on spec/views/index.html_spec.rb:
```diff
@@ -23,5 +23,7 @@
     expect(@page.text).to include("Podcast")
     expect(@page.text).to include("ICYMI")
+    expect(@page.text).to include("callout")
+    expect(@page.text).to include("Sponsors")
```
Let's add specs that correspond to the breakout methods.

jonathanyeong replied:
Okay will do.

sarony commented on lib/html_factory.rb:
```diff
@@ -709,39 +701,26 @@ def content(content_lists, callouts, sponsors)
-       </div>
-      DOC
-      string << content_string 
+      string << picture_post(post)
```
very nice

sarony commented on lib/html_factory.rb:
```diff
@@ -760,4 +739,33 @@ def content(content_lists, callouts, sponsors)
   end
+
+  def section_title(html_class, title, has_hr)
+    if has_hr
```
is "Join Us" the only one with no hr? If so, it might make more sense to have a default value of true.

jonathanyeong replied:
Yeah it is. That's a good idea. I'll make the change.

sarony commented:
@jonathanyeong made some minor comments. Once you take a look and respond, ping me and I'll take another look.

jonathanyeong committed:
Move fixtures into the spec folder

jonathanyeong committed:
Set default value for has_hr

The only method that doesn't use hr is the join us title.

jonathanyeong committed:
Add tests for break out methods

jonathanyeong committed:
Set correct path for fixtures

The fixture was being generated in the root project directory. Move this
to the spec folder.

jonathanyeong committed:
Write test for callout content method

jonathanyeong committed:
Uncomment out print messages

These messages give you an update on progress.

jonathanyeong committed:
Filter sensitive API keys in the VCR cassette
